lib/spack/spack/cmd/develop.py

```diff
@@ -85,8 +85,8 @@ def _retrieve_develop_source(spec: spack.spec.Spec, abspath: str) -> None:
 
 
 def develop(parser, args):
+    env = spack.cmd.require_active_env(cmd_name="develop")
```

Comment:
This line appears several additional times in this command, can you clean that up now that this is enforced at the top?

Reply:
Consolidated and moved the comments on the open PR. I'm question the desire for develop specs in a custom scope that is not an environment, but I preserved the intent in case I'm wrong.